adapters/repos/db/vector/hnsw/distancer/dot_product.go

```diff
@@ -30,12 +29,21 @@ var dotProductImplementation func(a, b []float32) float32 = func(a, b []float32)
 	return sum
 }
 
-func DotProductGo[T constraints.Integer | constraints.Float](a, b []T) T {
-	var sum T
+func DotProductFloatGo(a, b []float32) float32 {
+	var sum float32
 	for i := range a {
 		sum += a[i] * b[i]
 	}
 
+	return -sum
+}
```

Comment:
hmm, you can leave the generic `DotProduct` method and make it like that:
```
func DotProductFloatGo(a, b []float32) float32 {
	return -dotProductGo[float32, float32](a, b)
}

func DotProductByteGo(a, b []uint8) uint32 {
	return dotProductGo[uint8, uint32](a, b)
}

func dotProductGo[C uint8 | float32, T uint32 | float32](a, b []C) T {
	var sum T
	for i := range a {
		sum += T(a[i]) * T(b[i])
	}
	return sum
}
```

Reply:
Good point, change made